lib/Logitrail/Lib/ApiClient.php

```diff
@@ -3,327 +3,334 @@
+	/**
+	 * Confirm a passive order reported earlier for delivery
+	 *
+	 * @param string $logitrailOrderId
+	 * @return array The response resturned by Logitrail
+	 */
+	public function confirmOrder($logitrailOrderId) {
+		return $this->doPost($this->apiUrl . 'orders/' . $logitrailOrderId . '/_confirm');
+	}
+
+	/**
+	 * Create products to Logitrail's system.
+	 * Creates all products that were added with addProduct method and returns
+	 * array of responses from Logitrail.
+	 *
+	 * @return array Responses from Logitrail for each added product
+	 */
+	public function createProducts() {
+		// TODO: add support for subproducts
+		$results = array();
+
+		foreach ($this->products as $id => $product) {
+			$productData = array(
+				'merchants_id' => $product['id'],
+				'name'         => $product['name'],
+				'gtin'         => $product['barcode'],
+				'weight'       => $product['weight'],
+				'dimensions'   => array(
+					$product['width'],
+					$product['height'],
+					$product['length']
+				)
+			);
+
+			$results[] = $this->doPost($this->apiUrl . 'products/', $productData);
+		}
+
+		return $results;
+	}
+
+	/**
+	 * Update products to Logitrail's system.
+	 * Updatess all products that were added with addProduct method and returns
+	 * array of responses from Logitrail.
+	 *
+	 * Products are matched for updating by id.
+	 *
+	 * @return array Responses from Logitrail for each added product
+	 */
+	public function updateProducts() {
+		// Convenience function to keep method naming logical
+		// Create and update go to same endpoint in Logitrail and work the same way
+
+		return $this->createProducts();
+	}
+
+	/**
+	 * Remove products from the instance
+	 */
+	public function clearProducts() {
+		$this->products = array();
+	}
+
+	/**
+	 * Remove customer info from the instance
+	 */
+	public function clearCustomerInfo() {
+		$this->firstName  = NULL;
+		$this->lastName   = NULL;
+		$this->address    = NULL;
+		$this->postalCode = NULL;
+		$this->city       = NULL;
+		$this->phone      = NULL;
+		$this->email      = NULL;
+	}
+
+	/**
+	 * Remove order id from instance
+	 */
+	public function clearOrderId() {
+		$this->orderId = NULL;
+	}
+
+	/**
+	 * Remove customer info, order id and products from the instance
+	 */
+	public function clearAll() {
 		$this->clearCustomerInfo();
 		$this->clearOrderId();
 		$this->clearProducts();
-    }
-
-
+	}
 
-    /**
-     * Does a post call to Logireail's system to given endpoint with optional payload
-     *
-     * @param string $url URL of the endpoint to post to
-     * @param array $data Data sent as JSON payload
-     * @return array The response resturned by Logitrail
-     */
-    private function doPost($url, $data = false) {
-		if(!$this->merchantId || !$this->secretKey) {
+	/**
+	 * Does a post call to Logireail's system to given endpoint with optional payload
+	 *
+	 * @param string     $url  URL of the endpoint to post to
+	 * @param array|bool $data Data sent as JSON payload
```

Comment:
We don't wanna get bolean when this is used, but only array on nothing. Thus the correct change would be to put empty array as default and remove bool from docblock.